bagisto/src/main/java/shop/pageobjects/verifyShoppingCartProduct.java

```diff
@@ -0,0 +1,155 @@
+	
+	@FindBy(css="span[class*=\"card-current-price \"]")
+	List<WebElement> subTotal;
+	
+	@FindBy(css="div[class*=\"order-summary\"]")
+	WebElement cartSummary;
+
+	@FindBy (xpath="//a[text()=\"Proceed to checkout\"]")
+	WebElement checkoutButton;
+	
+	// use following sibling locator to grab grandTotal and discountPrice
+	
+	@FindBy(xpath="//span[text()=\"Sub Total\"]/following-sibling::span")
+	WebElement subtotal;
+	
+	@FindBy (xpath="//span[text()=\"Amount discounted\"]/following-sibling::span")
+	WebElement discount;
+
+	@FindBy (xpath="//span[text()=\"Grand Total\"]/following-sibling::span")
+	WebElement grandTotal;
+	
+	@FindBy (css="div[class*=\"address-container\"] div[class=\"card\"]")
+	List<WebElement> shipAddress;
+	
+	@FindBy (xpath="//input[@type=\"radio\" and @name=\"billing[address_id]\"]")
+	List<WebElement> shipAddressRadio;
+	
+	@FindBy (css="div[class*=\"shipping-methods \"] input[name=\"shipping_method\"]")
+	List<WebElement> shipMethod;
+	
+	@FindBy (xpath="//input[@type=\"radio\" and @name=\"shipping_method\"]")
+	List<WebElement> shippingMethodRadio;
+	
+	@FindBy (css="div[class=\"payment-methods\"] input[name=\"payment[method]\"]")
+	List<WebElement> paymentMethod;
+	
+	@FindBy (css="button[id=\"checkout-place-order-button\"]")
+	WebElement placeOrder;
+	
+	@FindBy (css="div[class*=\"order-success-content\"] a")
+	WebElement orderSuccess;
+	
+	By order = By.cssSelector("div[class*=\"order-success-content\"] a");
+	
+	@FindBy (css="div[class*=\"order-success-content\"]")
+	WebElement orderComplete;
+	
+	@FindBy (css="div[class*=\"order-success-content\"] h1")
+	WebElement thankyouOrder;
+	
+	@FindBy (css="div[class*=\"item-attribute\"]")
+	List<WebElement> bundleQuantity;
+	
+	By cartSubTotal = By.xpath("//span[text()=\"Sub Total\"]");
+	
+    By productPrice = By.cssSelector("div[class=\"product-price\"]");
+    
+	By productName = By.cssSelector("div[class*=\"item-title\"]");
+	
+	By spanTag = By.tagName("span");
+	
+	/* Vew Cart page, product quantity
+	 *  Price , subtotal , discount, GrandTotal */
+    public void verifyShoppingCart() throws InterruptedException {
+    	ProductActionHelper action = new ProductActionHelper(driver);
+    	action.goToCartPage();
+	   float expectedCartSubTotal= 0 ;
+	   int quantity=0;
+	   for (int i=0;i<totalProductOnCart.size();i++) {
+		  float price= remove$(totalProductOnCart.get(i).findElement(productPrice).getText());
+		  if(productQuantity.size()==0) {
+			  quantity = bundleQuantity.size();
+		  }
+		  else {
+		      quantity = Integer.parseInt(productQuantity.get(i).getAttribute("model"));
+		  }
+		  float expectedProductSubTotal=0;
+		  float actualProductSubTotal = remove$(subTotal.get(i).getText());
+		  System.out.println("Product In cart: "+totalProductOnCart.get(i).findElement(productName).getText());
+		  System.out.println("Product Price: "+price);
+		  System.out.println("Product Quantity: "+quantity);
+		  expectedProductSubTotal = price*quantity;
```

Comment:
This will give an error if the quantity box is not available for any product. this code will execute only if the quantity box is visible

Reply:
Changed code according to condition, when qunatity is not available